### PR TITLE
Redesign question bank

### LINKED ISSUE #3030: CiviForm admin UX: hide question bank and add "add question" button
**Is your feature request related to a problem? Please describe.**
When creating a program, the CiviForm admin is confused/overwhelmed by the long list of questions in the question bank 

**Describe the solution you'd like**
Do not show the question bank on the "manage questions" page when a user first lands on the page. Instead, add a button that says "add question" to each page. When a user clicks the button, then the question bank appears 

**Additional context**
From admin UXR feedback sheet

**Screenshot**
(ignore the "allow user to reorder question" text on the screenshot. That's github issue #1928 )
Screen Shot 2022-08-02 at 3 59 06 PM

### PR DESCRIPTION
**Pull request by nbeloglazov**
### Description

Change question bank to be hidden by default. Each program screen has a "Add question" button now. When you clicks it - question bank appears where user can add questions from. See video:

https://drive.google.com/file/d/1xERLRLzJ32ky0eALqlZAP5XWHGaZ5cyw/view?usp=sharing

Try it out: https://staging-mikita.civiform.dev/

This is a pre-requisite for bigger update to program edit page: https://github.com/civiform/civiform/issues/3419 

## Release notes:

Redesign question bank to be hidden by default. It's now triggered by a new "Add a question" button. 

### Checklist

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)

Fixes #3030.

### CODE REVIEW

File: browser-test/src/support/admin_programs.ts [hunk 1]
```diff
@@ -312,11 +312,26 @@ export class AdminPrograms {
     }
   }
 
+  async openQuestionBank() {
+    await this.page.click('button:has-text("Add a question")')
+    // wait for animation to finish
+    await this.page.waitForTimeout(1000)
```
Comment (bion) [inline browser-test/src/support/admin_programs.ts:318]: suggestion (blocking): if the animation is 1s please extract 1000 here to a constant

File: server/app/assets/javascripts/questionBank.ts [hunk 2]
```diff
@@ -12,10 +16,67 @@ class QuestionBankController {
...
+    // "add question" button we want to display questyion bank using quick animation.
+    // But CSS transitions don't work if they are triggered simultaneously with display:
+    // https://stackoverflow.com/questions/3331353/transitions-on-the-css-display-property
+    // To workaround we use 2 classes: `hidden` and `cf-question-bank-hidden`.
+    //
+    // When showing question we first remove `hidden` and wait for browser to apply it by
+    // using requestAnimationFrame() and then remove `cf-question-bank-hidden` class.
+    //
+    // When hiding we first add `cf-question-bank-hidden` to trigger transition and once
+    // it's over add `hidden` class.
```
Comment (bion) [inline server/app/assets/javascripts/questionBank.ts:37]: great work getting this sorted out!

**Review by bion — CHANGES_REQUESTED** [review]
Great work! This is such a huge improvement!

One small bug one testing: refreshing the page after initially opening the "add question" sidebar does not keep the sidebar open which is inconsistent with refreshing that page after adding a question. Looks like opening the side bar should also update the URL to have `sqb=true`

File: browser-test/src/support/admin_programs.ts (hunk 1 above)
Comment (nbeloglazov) [reply browser-test/src/support/admin_programs.ts:318]: Animation is 150ms: https://tailwindcss.com/docs/transition-property

I put 1s as upper limit to hopefully avoid flakiness. I can expand comment here. But don't see how having it as constant would be useful yet.

File: server/app/views/components/QuestionBank.java [hunk 3]
```diff
@@ -249,6 +288,19 @@ private Optional<Long> getEnumeratorQuestionId() {
     return enumeratorQuestionId;
   }
 
+  /**
+   * Question bank is hidden by default when user opens program edit page. But some actions, like
+   * adding question from the bank, lead to page reload and after such reload question bank should
+   * stay open as that matches user expectations.
+   */
+  public static String addShowQuestionBankParam(String url) {
+    return url + (url.contains("?") ? "&" : "?") + QuestionBank.SHOW_QUESTION_BANK_PARAM + "=true";
```
[the remark this answers is not in the document]
Comment (nbeloglazov) [reply server/app/views/components/QuestionBank.java]: Thanks! Didn't realize we have URIBuilder available. Agree that's much better.

File: browser-test/src/support/admin_programs.ts (hunk 1 above)
Comment (bion) [reply browser-test/src/support/admin_programs.ts:318]: Ah got it, a comment would be good then

File: server/app/views/components/QuestionBank.java [hunk 4]
```diff
@@ -48,24 +46,69 @@ public QuestionBank(QuestionBankParams params) {
     this.params = checkNotNull(params);
   }
 
-  public FormTag getContainer() {
-    return questionBankPanel();
+  public DivTag getContainer(boolean isQuestionBankVisible) {
```
[the remark this answers is not in the document]
Comment (nbeloglazov) [reply server/app/views/components/QuestionBank.java]: Done.

File: browser-test/src/support/admin_programs.ts (hunk 1 above)
Comment (nbeloglazov) [reply browser-test/src/support/admin_programs.ts]: Done.

**Review by bion — APPROVED** [review]

**Pull request merged**